server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeDataLoader.java

```diff
@@ -164,24 +152,11 @@ private Long searchDeveloperId(DbSession dbSession, ComponentTreeWsRequest wsReq
       .uniqueIndex(ComponentDto::uuid);
   }
 
-  private ComponentDtosAndTotal searchComponents(DbSession dbSession, ComponentTreeQuery dbQuery, ComponentTreeWsRequest wsRequest) {
-    if (dbQuery.getQualifiers() != null && dbQuery.getQualifiers().isEmpty()) {
-      return new ComponentDtosAndTotal(Collections.emptyList(), 0);
-    }
-    String strategy = requireNonNull(wsRequest.getStrategy());
-    switch (strategy) {
-      case CHILDREN_STRATEGY:
-        return new ComponentDtosAndTotal(
-          dbClient.componentDao().selectChildren(dbSession, dbQuery),
-          dbClient.componentDao().countChildren(dbSession, dbQuery));
-      case LEAVES_STRATEGY:
-      case ALL_STRATEGY:
-        return new ComponentDtosAndTotal(
-          dbClient.componentDao().selectDescendants(dbSession, dbQuery),
-          dbClient.componentDao().countDescendants(dbSession, dbQuery));
-      default:
-        throw new IllegalStateException("Unknown component tree strategy");
+  private List<ComponentDto> searchComponents(DbSession dbSession, ComponentTreeQuery componentTreeQuery) {
+    if (componentTreeQuery.getQualifiers() != null && componentTreeQuery.getQualifiers().isEmpty()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) NullPointerException might be thrown as 'getQualifiers' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)